### PR TITLE
feat(db): add entity_checks for /schema endpoint

### PR DESCRIPTION
**Pull request by jschmid1**
### Summary


The `/schemas/{plugins,$daos}` endpoint now not only returns fields, but also any canonical entity_checks that can't be expressed otherwise within the field validators themselves.

We do **not** include `custom_entity_checks` that usually implement custom lua functions to verify the validity.

For example:

`http :8001/schemas/keys`



``` json
// shortened for brevity
{                                                                                                                                                           
    "entity_checks": [                                                                                                                                      
        {                                                                                                                                                   
            "mutually_exclusive": [                                                                                                                         
                "jwk",                                                                                                                                      
                "pem"                                                         
            ]                                                                                                                                               
        },                                                                                                                                                                                                                                                                
    ],                                                                        
    "fields": [                                                                                                                                             
        {                                                                                                                                                   
            "id": {                                                                                                                                         
                "auto": true,                                                                                                                               
                "description": "A string representing a UUID (universally unique identifier).",                                                             
                "type": "string",                                                                                                                           
                "uuid": true                                                                                                                                
            },                                                                                                                                               
        },
     ],                 
}

```

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG


### Full changelog

- Add `entity_checks` to /schemas endpoint output

### Issue reference

https://konghq.atlassian.net/browse/KAG-1593

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 #### Core
 
+- Add `entity_checks` to /schemas endpoint output
```
Comment (hbagdi) [inline CHANGELOG.md:22]:
```suggestion
- '/schemas' endpoint returns additional information about cross-field validation as part of the schema. This should help tools that use the Admin API to perform better client-side validation. 
```

A good changelog entry always describes the change from a user's perspective. Once again, the changelog is meant for users and not maintainers.

File: spec/02-integration/04-admin_api/04-plugins_routes_spec.lua [hunk 2]
```diff
@@ -450,11 +451,12 @@ for _, strategy in helpers.each_strategy() do
             path = "/schemas/plugins/request-transformer",
           })
           local body = assert.res_status(200, res)
-          assert.match('{"fields":[{', body, 1, true)
+          assert.not_match('"entity_checks":{', body, 1, true)
           assert.not_match('"fields":{', body, 1, true)
           assert.match('"default":[]', body, 1, true)
           assert.not_match('"default":{}', body, 1, true)
         end)
+
```
Comment (hbagdi) [inline spec/02-integration/04-admin_api/04-plugins_routes_spec.lua:459]: unnecessary line jump

File: spec/02-integration/04-admin_api/04-plugins_routes_spec.lua [hunk 3]
```diff
@@ -442,6 +442,7 @@ for _, strategy in helpers.each_strategy() do
             local body = assert.res_status(200, res)
             local json = cjson.decode(body)
             assert.is_table(json.fields)
+            assert.is_table(json.entity_checks)
```
Comment (hbagdi) [inline spec/02-integration/04-admin_api/04-plugins_routes_spec.lua:445]: Please assert the content of the new entry.
Please add a negative test to make sure that 'entity_checks' JSON key is not present when it is absent from the Lua schema.

File: spec/02-integration/04-admin_api/04-plugins_routes_spec.lua [hunk 4]
```diff
@@ -450,11 +451,12 @@ for _, strategy in helpers.each_strategy() do
             path = "/schemas/plugins/request-transformer",
           })
           local body = assert.res_status(200, res)
-          assert.match('{"fields":[{', body, 1, true)
+          assert.not_match('"entity_checks":{', body, 1, true)
```
Comment (hbagdi) [inline spec/02-integration/04-admin_api/04-plugins_routes_spec.lua:454]: I'm not sure how to read this change.
Should the code not assert that the 'fields' JSON key is present and 'entity_checks' is not present?

**Review by hbagdi — CHANGES_REQUESTED** [review]

File: spec/02-integration/04-admin_api/04-plugins_routes_spec.lua (hunk 3 above)
Comment (jschmid1) [reply spec/02-integration/04-admin_api/04-plugins_routes_spec.lua:445]: I'd argue that we always return `entity_checks` in the JSON response even when the value is empty for the sake of consistency.

File: spec/02-integration/04-admin_api/04-plugins_routes_spec.lua (hunk 4 above)
Comment (jschmid1) [reply spec/02-integration/04-admin_api/04-plugins_routes_spec.lua:454]: This is a follow-up based on [this assumption](https://github.com/Kong/kong/pull/11108/files#r1244201136). 

When we always include `entity_check` regardless of its contents, this test is verifying if a key named `entity_checks`, is present (analogous to the other assertions)

**Review by fabianrbz — APPROVED** [review]
LGTM

**Pull request merged**